### PR TITLE
Features/RemoteConfig

### PR DESCRIPTION
**Pull request by nkashyap**
## RemoteConfig

This PR implements FIrebase remote-config api for both Android & IOS

### TODO checklist

- [x] Implementation - JS
- [x] Implementation - Android
- [x] Implementation - IOS
- [x] Documentation
- [ ] Unit tests - JS
- [ ] Unit tests - Android
- [ ] Unit tests - IOS
- [x] Cross Platform check
- [ ] Update - FirestackApp

### CODE REVIEW

File: android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        }
+    }
+
+    @ReactMethod
+    public void fetch(final Integer cacheExpirationSeconds, final Promise promise) {
+        try {
+            Task fetchTask = mFirebaseRemoteConfig.fetch(cacheExpirationSeconds.longValue());
+            fetchTask.addOnCompleteListener(new OnCompleteListener<Void>() {
+                @Override
+                public void onComplete(@NonNull Task<Void> task) {
```
Comment (achintkalra) [inline android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java]: If using Java 8, consider using lambda

File: android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+            promise.reject("Exception", e.getMessage(), e);
+        }
+    }
+
+    @ReactMethod
+    public void isDeveloperModeEnabled(Promise promise) {
+        try {
+            promise.resolve(mFirebaseRemoteConfig.getInfo().getConfigSettings().isDeveloperModeEnabled());
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (achintkalra) [inline android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java]: Is printStackTrace required?

File: android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java (hunk 1 above)
Comment (nkashyap) [reply android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java]: @achintkalra thanks Achint, This code will run in Android, will that make any difference?

File: android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+            promise.reject("Exception", e.getMessage(), e);
+        }
+    }
+
+    @ReactMethod
+    public void getKeysByPrefix(String prefix, String namespace, Promise promise) {
+        try {
+            Set<String> value;
+
+            if (namespace != null && !namespace.isEmpty()) {
```
Comment (achintkalra) [inline android/src/main/java/io/fullstack/firestack/remoteconfig/FirestackRemoteConfig.java]: Consider using Apache's StringUtils.isBlank

File: ios/Firestack/FirestackRemoteConfig.m [hunk 4]
```diff
@@ -0,0 +1,232 @@
+//
+//  FirestackRemoteConfig.m
+//  Firestack
```
Comment (nkashyap) [inline ios/Firestack/FirestackRemoteConfig.m:3]: @Salakar @Ehesp @chrisbianca  Hello guys, please can you help me with objective-c code.

Comment (nkashyap) [reply ios/Firestack/FirestackRemoteConfig.m:3]: I managed to put some objective-c code together using example, but it is throwing exceptions.

File: ios/Firestack.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -290,11 +303,14 @@
...
 				);
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				LIBRARY_SEARCH_PATHS = "$(inherited)";
 				MACH_O_TYPE = staticlib;
 				OTHER_LDFLAGS = "$(inherited)";
 				PRODUCT_NAME = Firestack;
 				SKIP_INSTALL = YES;
+				SWIFT_OBJC_BRIDGING_HEADER = "Firestack-Bridging-Header.h";
+				SWIFT_VERSION = 3.0;
```
Comment (nkashyap) [inline ios/Firestack.xcodeproj/project.pbxproj]: remove swift reference

File: ios/Firestack.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -261,19 +269,24 @@
...
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				LIBRARY_SEARCH_PATHS = "$(inherited)";
 				MACH_O_TYPE = staticlib;
 				ONLY_ACTIVE_ARCH = YES;
 				OTHER_LDFLAGS = "$(inherited)";
 				PRODUCT_NAME = Firestack;
 				SKIP_INSTALL = YES;
+				SWIFT_OBJC_BRIDGING_HEADER = "Firestack-Bridging-Header.h";
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 3.0;
```
Comment (nkashyap) [inline ios/Firestack.xcodeproj/project.pbxproj]: remove swift references

**Pull request closed without merging**